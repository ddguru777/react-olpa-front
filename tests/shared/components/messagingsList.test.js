/**
 * Copyright (c) 2015-present, CWB SAS
 *
 * This source code is licensed under the GPL v2.0+ license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from "react";
import renderer from "react-test-renderer";
import MessagingsList from "shared/components/messagingsList";

describe("components/messagingsList", () => {
  it("renders correctly", () => {
    const item = [
      { name: "foo", icon: "favorite" },
      { name: "bar", icon: "foobar", enable: "true" },
      { name: "title", icon: "" },
    ];
    const component = renderer.create(
      <MessagingsList name="foo" items={item} onSelect={jest.fn()} />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
