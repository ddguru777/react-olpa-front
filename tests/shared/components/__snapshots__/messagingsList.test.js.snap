// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/messagingsList renders correctly 1`] = `
<div
  className="zui-sublist"
>
  <div
    className="zui-subheader"
  >
    <h4>
      foo
    </h4>
  </div>
  <ul
    className="mdc-list"
  >
    <li
      className="selectableListItem switchListItem mdc-list-item"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="menuitem"
      style={
        Object {
          "height": "40px",
        }
      }
    >
      <span
        className="mdc-list-item__text"
      >
        foo
        <span
          className="mdc-list-item__secondary-text"
        >
          click to setup
        </span>
      </span>
      <span
        className="mdc-list-item__meta"
        onClick={null}
        onKeyUp={[Function]}
        role="button"
        tabIndex="0"
      >
        <div
          className="mdc-switch"
          onClick={[Function]}
        >
          <div
            className="mdc-switch__track"
          />
          <div
            className="mdc-switch__thumb-underlay"
          >
            <div
              className="mdc-switch__thumb"
            >
              <input
                checked={false}
                className="mdc-switch__native-control"
                disabled={false}
                id="unique-component-id"
                role="switch"
                type="checkbox"
              />
            </div>
          </div>
        </div>
      </span>
    </li>
    <li
      className="selectableListItem switchListItem mdc-list-item"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="menuitem"
      style={
        Object {
          "height": "40px",
        }
      }
    >
      <span
        className="mdc-list-item__text"
      >
        bar
        <span
          className="mdc-list-item__secondary-text"
        >
          click to setup
        </span>
      </span>
      <span
        className="mdc-list-item__meta"
        onClick={null}
        onKeyUp={[Function]}
        role="button"
        tabIndex="0"
      >
        <div
          className="mdc-switch"
          onClick={[Function]}
        >
          <div
            className="mdc-switch__track"
          />
          <div
            className="mdc-switch__thumb-underlay"
          >
            <div
              className="mdc-switch__thumb"
            >
              <input
                checked={false}
                className="mdc-switch__native-control"
                disabled={false}
                id="unique-component-id"
                role="switch"
                type="checkbox"
              />
            </div>
          </div>
        </div>
      </span>
    </li>
    <li
      className="selectableListItem switchListItem mdc-list-item"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="menuitem"
      style={
        Object {
          "height": "40px",
        }
      }
    >
      <span
        className="mdc-list-item__text"
      >
        title
        <span
          className="mdc-list-item__secondary-text"
        >
          click to setup
        </span>
      </span>
      <span
        className="mdc-list-item__meta"
        onClick={null}
        onKeyUp={[Function]}
        role="button"
        tabIndex="0"
      >
        <div
          className="mdc-switch"
          onClick={[Function]}
        >
          <div
            className="mdc-switch__track"
          />
          <div
            className="mdc-switch__thumb-underlay"
          >
            <div
              className="mdc-switch__thumb"
            >
              <input
                checked={false}
                className="mdc-switch__native-control"
                disabled={false}
                id="unique-component-id"
                role="switch"
                type="checkbox"
              />
            </div>
          </div>
        </div>
      </span>
    </li>
  </ul>
</div>
`;
